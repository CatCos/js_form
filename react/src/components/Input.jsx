import React from 'react';
import PropTypes from 'prop-types';
import classSet from 'classnames';

import Tooltip from './Tooltip.jsx';

function handleChange(e, inputName, props) {
  if (props.onChange) {
    return props.onChange({
      e: e,
      inputName: inputName
    });
  }
}

function renderTooltip(props) {
  return (
    <Tooltip value={props.error}/>
  );
}

/**
 * Render function
 *
 * @params props Component properties
 */
function Input(props) {
  let className = [props.className];

  if (props.hasError) {
    className.push('input-error');

    className = classSet(className);
  }

  return (

    <div className="form-group">
        <label htmlFor={props.name} className="label">{props.text}</label>
        <input
          className={className}
          id={props.name}
          type={props.type}
          value={props.value}
          name={props.name}
          onChange={(e) => handleChange(e, props.field, props)}/>
        {props.hasError && renderTooltip(props)}
    </div>
  );
}

Input.propTypes = {
  className: PropTypes.string,
  name: PropTypes.string,
  type: PropTypes.string,
  field: PropTypes.string,
  error: PropTypes.string,
  text: PropTypes.string,
  onChange: PropTypes.func,
  hasError: PropTypes.bool
};

Input.defaultProps = {
  name: '',
  value: '',
  field: '',
  error: '',
  text: '',
  className: 'input',
  hasError: false,
  type: 'text'
};

export default Input;
