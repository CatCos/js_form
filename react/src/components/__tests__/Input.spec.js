import React from 'react';
import Input from '../Input';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <Input
      value="text"
      name='text'
      text='text input'
      onChange={jest.fn()}
      text='Submit comment'/>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
