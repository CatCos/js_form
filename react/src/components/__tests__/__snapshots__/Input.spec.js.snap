// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="form-group"
>
  <label
    className="label"
    htmlFor="text"
  >
    Submit comment
  </label>
  <input
    className={
      Array [
        "input",
      ]
    }
    id="text"
    name="text"
    onChange={[Function]}
    type="text"
    value="text"
  />
</div>
`;
